packages/okta-vue/README.md

```diff
@@ -0,0 +1,223 @@
+
+```bash
+npm install --save @okta/okta-vue
+```
+
+### Configuration
+You will need the values from the OIDC client that you created in the previous step to instantiate the middleware. You will also need to know your Okta Org URL, which you can see on the home page of the Okta Developer console.
+
+In your application's [vue-router](https://router.vuejs.org/en/essentials/getting-started.html) configuration, import the `@okta/okta-vue` plugin and pass it your OpenID Connect client information:
+
+```typescript
+// router/index.js
+
+import Auth from '@okta/okta-vue'
+
+Vue.use(Auth, {
+  issuer: 'https://{yourOktaDomain}.com/oauth2/default',
+  client_id: '{client_id}',
+  redirect_uri: 'http://localhost:{port}/implicit/callback',
+  scopes: ['openid', 'profile', 'email']
+})
+
+```
+
+### Use the Callback Handler
+In order to handle the redirect back from Okta, you need to capture the token values from the URL. You'll use `/implicit/callback` as the callback URL, and use the default `Auth.handleCallback()` component included.
+
+```typescript
+// router/index.js
+
+const router = new Router({
+  ...
+  mode: 'history',
+  routes: [
+    { path: '/implicit/callback', component: Auth.handleCallback() },
+    ...
+  ]
+})
+```
+
+### Add a Protected Route
+Routes are protected by the `authRedirectGuard`, which verifies there is a valid `accessToken` or `idToken` stored. To ensure the user has been authenticated before accessing your route, add the `requiresAuth` metadata:
+
+```typescript
+// router/index.js
+
+{
+  path: '/protected',
+  component: Protected,
+  meta: {
+    requiresAuth: true
+  }
+}
+```
+
+Next, overload your router's `beforeEach()` executer to inject the global [navigation guard](https://router.vuejs.org/en/advanced/navigation-guards.html):
+
+```typescript
+// router/index.js
+
+router.beforeEach(Vue.prototype.$auth.authRedirectGuard())
+```
+
+If a user does not have a valid session, they will be redirected to the Okta Login Page for authentication. Once authenticated, they will be redirected back to your application's **protected** page.
+
+### Show Login and Logout Buttons
+In the relevant location in your application, you will want to provide `Login` and `Logout` buttons for the user. You can show/hide the correct button by using the `$auth.isAuthenticated()` method. For example:
+
+```typescript
+// src/App.vue
+
+<template>
+  <div id="app">
+    <router-link to="/" tag="button" id='home-button'> Home </router-link>
+    <button v-if='authenticated' v-on:click='logout' id='logout-button'> Logout </button>
+    <button v-else v-on:click='$auth.loginRedirect' id='login-button'> Login </button>
+    <router-view/>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'app',
+  data: function () {
+    return { authenticated: false }
+  },
+  created () { this.isAuthenticated() },
+  watch: {
+    // Everytime the route changes, check for auth status
+    '$route': 'isAuthenticated'
```

Comment:
Watching '$route' wasn't causing any hooks for me?  I ended up adding a call to `this.isAuthenticated()` below `await this.$auth.logout()`

Reply:
We need both in this case. On route change, we need to call `this.$auth.isAuthenticated()` to ensure the DOM is updated correctly. Looks like `logout` had some issues and are fixed in the next commit.